### ISSUE TITLE
Edit option closeAfterEdit not functioning

### ISSUE THREAD

**Issue by bchr02** [openpsa/jsgrid#104]
Edit option closeAfterEdit not functioning.

````javascript
$("#grid").jqGrid(
    'navGrid',
    '#pager',
    {add: false, edit: true, del: false}, // navGrid options
    {closeAfterEdit: true}  // Edit options
);
````
However it does work when placed in
````javascript
$.extend($.jgrid.edit, {closeAfterEdit: true}); 
````

**Comment by bouks** [comment]
Working good on my project.
In your example, closeAfterEdit is in add button options.

try

````javascript
$("#grid").jqGrid(
    'navGrid',
    '#pager',
    {add: false, edit: true, del: false}, // navGrid options
    {},
    {closeAfterEdit: true}  // Edit options
);

**Comment by bouks** [comment]
Or

````javascript
$("#grid").jqGrid(
    'navGrid',
    '#pager',
    {edit: true, add: false, del: false}, // navGrid options
    {closeAfterEdit: true}  // Edit options
);

**Comment by flack** [comment]
@bouks: I just looked at the function signature here:

https://github.com/openpsa/jsgrid/blob/6c5783eeea53c0a04337ec1ebbca3cfe07116b11/js/grid.formedit.js#L1742

so edit options should be the third parameter

**Comment by flack** [comment]
sorry, just saw your second post after submitting ...

**Comment by bouks** [comment]
@flack I think to avoid this type of issue we should probably in a future version (2.0?) implement like this :

````javascript
$("#grid").jqGrid(
    'navGrid',
    '#pager',
    {
        edit: {
            closeAfterEdit: true,
            afterSubmit:function(...){
                ...
            }
        },
        add: false,
        del: false
    }
);
````

or without options


````javascript
$("#grid").jqGrid(
    'navGrid',
    '#pager',
    {
        edit: true,
        add: false,
        del: false
    }
);
````

**Comment by bouks** [comment]
Then, no matter the orders.

**Comment by flack** [comment]
@bouks: Yes, the API can definitely be improved. But to return to the original issue: Do I understand it correctly that the example from @bchr02 works on your machine?

**Comment by bouks** [comment]
@flack Yes.

**Comment by bouks** [comment]
But i do a different implementation of navgrid in my projects.
I do a function (that call navgrid with the parameters) and i call this function in gridComplete option.
What i remember is that i had problems with adding buttons in pager with conventionnal call.

**Comment by bchr02** [comment]
We should definitely improve the API. :+1: but in the meantime this appears to be a bug. It should be the third parameter, correct?

**Comment by bouks** [comment]
Are you using server side data ?
What is the return of the ajax call when you validate the form ?

**Comment by bouks** [comment]
If the return is a 300 code, or a 304 code (Not Modified), the closeAfterEdit won't happen.
Could you check this ?

**Comment by bchr02** [comment]
It looks something like this. { rowsAffected: undefined, outBinds: {}, rows: undefined }
and it's a 200 code

**Comment by bouks** [comment]
Excuse me i'm wrong. It will not close if  code >= 300 AND code !== 304

**Comment by bchr02** [comment]
okay that makes sense but that is not the case here.

**Comment by bouks** [comment]
What is the returned data of your ajax call ?
Do you have an afterSubmit function defined in your edit button options ?

**Comment by bchr02** [comment]
no afterSubmit defined.

Here is the returned data { rowsAffected: 1, outBinds: {}, rows: undefined }

**Comment by flack** [comment]
@bchr02 I just tried to reproduce this locally, but for me, `closeAfterEdit` works fine. I've made a demo:

http://openpsa.github.io/jsgrid/demos/tickets/104.html

Unfortunately, it doesn't work on gh-pages, because they don't allow post requests, but if you run it on a local webserver, everything is fine. Do you have a demo where we can see the problem?

**Comment by bchr02** [comment]
I think the issue only happens when double clicking on a row. It may be because of how I implemented that. 
````javascript
ondblClickRow: function(rowid) {
	$(this).jqGrid('editGridRow', rowid);
},
````

When I get a chance I will create a demo but for now there is another issue that I have found that is more important. I will create a separate entry for that shortly.

**Comment by bchr02** [comment]
I got it working by changing it to this:
````javascript
ondblClickRow: function(rowid) {
	$(this).jqGrid('editGridRow', rowid,
		{	recreateForm:true,
			closeAfterEdit:true,
			closeOnEscape:true,
			reloadAfterSubmit:false});
},
````

What I think is happening is the ondblClickRow function is getting defined before the closeAfterEdit:true is set. Since the default is closeAfterEdit:false, this is how it is defined. 

This is another enhancement we can add where we can do something like this:
````javascript
ondblClickRow: function(rowid) {
	var self = this;
	$(self).jqGrid('editGridRow', rowid, function() {
		return $(self).jqGrid('getEditProperties');
	});
},
````
of coarse **getEditProperties** does not exist but we could add it. (unless there is another method already that I am unaware of)

But instead of implementing it in this fashion we should make a property that we could set to easily implement double click editing. Something like this:
````javascript
$("#grid").jqGrid('navGrid','#pager',{},{closeAfterEdit: true, ondblClickEdit: true});
````

**Comment by meh-uk** [comment]
That sounds cool, but not immediately!

**Comment by bchr02** [comment]
Okay thanks.
